.github/workflows/reusable_dockerfile_pipeline.yml

```diff
@@ -140,7 +140,15 @@ jobs:
       # only push if the branch/PR is not generated from a fork. Even though
       # forks can't push, we still want to try and build the image to catch
       # bugs. For testing purposes we only need an amd64 image.
-      - name: Build and Push Docker Image amd64
+      - name: Build and Push Experimental Docker Image amd64
+        # This will run only when the event is a push to a branch that is not 'main'
+        # and the ref is not a release candidate or release tag.
+        if: >-
+          github.event_name == 'push' &&
+          github.ref != 'refs/heads/main' &&
+          !startsWith(github.ref, 'refs/tags/v') &&
+          !contains(github.ref, '-rc') &&
+          github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name
```

Comment:
This means it does not build on a fork, which should happen, to report back an error to the developer if something breaks.
Building it always and only pushing on non-forks results in that behavior.